selinux/bluechi.te

```diff
@@ -98,6 +99,26 @@ dbus_system_bus_client(bluechi_agent_t)
 
 init_status(bluechi_agent_t)
 
+
+########################################
+#
+# bluechi policy for unix domain sockets
+#
+type bluechi_var_run_t;
+files_pid_file(bluechi_var_run_t)
+init_sock_file(bluechi_var_run_t)
+mls_trusted_object(bluechi_var_run_t)
+
+manage_sock_files_pattern(bluechi_t, bluechi_var_run_t, bluechi_var_run_t)
+stream_connect_pattern(bluechi_agent_t, bluechi_var_run_t, bluechi_var_run_t, bluechi_t)
+unconfined_server_stream_connectto(bluechi_agent_t)
+
```

Comment:
I think this is fine as a first step. However, we're going to need more here eventually so that we can allow bluechi-agent in the QM to also access this.

Maybe we can wait a bit with that because the ideal approach for that is to add a qm_bluechi_agent_t role in the qm selinux policy, rather than having all of QM be able to talk to the host bluechi controller.

Reply:
> rather than having all of QM be able to talk to the host bluechi controller.

Wouldn't this setup already prevent that? If the socket gets mounted into QM's `/run/bluechi` directory and by installing the bluechi-selinux policy, the directory would get the `bluechi_var_run_t` label and we'd allow only `bluechi_agent_t` to connect to it?

> Maybe we can wait a bit with that because the ideal approach for that is to add a qm_bluechi_agent_t role in the qm selinux policy, rather than having all of QM be able to talk to the host bluechi controller.

This means we need to have a `qm_bluechi_agent_r` in the QM policy, then require it here and grant it access to `bluechi_var_run_t`? I'd move this out of this PR/task to and create a GH issue in QM (core or extras?) as well as here in BlueChi to implement this.

Reply:
Yes you can either allow qm_t to:

stream_connect_pattern(qm_t, bluechi_var_run_t, bluechi_var_run_t, bluechi_t)

Or create a new type.  I would go with the simpler version.  Just need to make sure bluechi_t protects itself from clients.

I don't see qm_t needing to be as locked down as qm_container_t, Bottom line customers should be containerizing their QM apps and these apps will have little privs.

Reply:
Yeah, either a custom agent+qm type, or give all of qm_t access to the bluechi socket.

Reply:
Since BlueChi might be used inside QM and BlueChi doesn't know anything about QM, it would make sense to me have this in the SELinux policy for QM - similar to how BlueChi uses HAProxy. But not sure how this would look like or if it makes sense. 
Opening a [QM issue](https://github.com/containers/qm/issues) to further discuss this (and maybe move it back to BlueChi later).

Reply:
Update: Created https://github.com/containers/qm/issues/677 to keep track of this and discuss the solution further.